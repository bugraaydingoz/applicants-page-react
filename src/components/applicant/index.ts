export { Applicant } from './Applicant'
export { ApplicantList } from './ApplicantList'
