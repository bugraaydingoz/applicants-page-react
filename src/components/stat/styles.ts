import styled from 'styled-components'

export const StatContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 110px;
  border-right: 1px solid var(--gray-300);

  &:last-child {
    border: none;
  }
`
