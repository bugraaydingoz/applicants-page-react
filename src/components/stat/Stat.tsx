import React from 'react'

// Styles
import { StatContainer } from './styles'
import { H4, Body2 } from '../shared/typography'

interface Props {
  title: string
  value: number
}

export function Stat({ title, value }: Props) {
  return (
    <StatContainer>
      <H4>{value}</H4>
      <Body2>{title}</Body2>
    </StatContainer>
  )
}
