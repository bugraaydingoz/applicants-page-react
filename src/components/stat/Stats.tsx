import React from 'react'
import { useTranslation } from 'react-i18next'

// Components
import { Stat } from './Stat'

// Styles
import { Flex } from '../shared/layout'

interface Props {
  total: number
  new: number
  viewed: number
  appointment: number
  others: number
}

export function Stats(stats: Props) {
  const { t } = useTranslation()
  return (
    <Flex>
      <Stat title={t('stats.total')} value={stats.total} />
      <Stat title={t('stats.new')} value={stats.new} />
      <Stat title={t('stats.viewed')} value={stats.viewed} />
      <Stat title={t('stats.appointment')} value={stats.appointment} />
      <Stat title={t('stats.others')} value={stats.others} />
    </Flex>
  )
}
