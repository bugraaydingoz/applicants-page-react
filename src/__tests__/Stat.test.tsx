import React from 'react'
import renderer from 'react-test-renderer'
import { Stat } from '../components/stat'

test('renders correctly', () => {
  const tree = renderer.create(<Stat title="title" value={50} />).toJSON()
  expect(tree).toMatchSnapshot()
})
