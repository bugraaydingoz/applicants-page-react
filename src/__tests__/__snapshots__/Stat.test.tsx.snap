// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm dqQXLp"
>
  <h4
    className="sc-AxhCb iIoMBe"
  >
    50
  </h4>
  <div
    className="sc-AxgMl duusKk"
  >
    title
  </div>
</div>
`;
